### ISSUE TITLE
Log format for errors is not correct

### ISSUE THREAD

**Issue by friedrichg** [cortexproject/auth-gateway#17]
Using https://github.com/cortexproject/auth-gateway/commit/2265b2bd464daa71dd374070bae4d6a0b84c94b4

$ ./auth-gateway gateway/testdata/configuration.yaml
server has started
**2023/04/23 18:26:21 http: proxy error: dial tcp 127.0.0.1:8081: connect: connection refused**
level=debug msg="No valid tenant credentials are found"

$ curl -u username1:password1 http://127.0.0.1:8080/api/prom/push
$ curl http://127.0.0.1:8080/api/prom/push

this error is sent by go default logger. we should only use logrus

We should modify [ErrorLog](https://github.com/golang/go/blob/go1.20.3/src/net/http/httputil/reverseproxy.go#L173-L176) in the ReverseProxy with logrus

**Comment by dogukanteber** [comment]
Yeah, this was the subtask of #8 as well but I focused too much on the specific error in that issue. I will fix it right away
